City-of-Helsinki/helsinki-paatokset
# PP-332: Improve load times, add lazybuilding methods for policymaker listing

This PR adds lazybuilding methods for the policymakers list and improves caching for the meetings calendar.

**To test**
* Checkout branch
* Modify your local.settings.php and comment out everything related to development services and caching
* Run `make composer-install drush-cr drush-cim`
* Test that the meetings calendar works as admin and anonymous user: https://helsinki-paatokset.docker.so/fi/kokouskalenteri
  *  It should load for a few seconds on the first page load (and display the loading icon), but on subsequent page loads it should be considerably faster
  * Check the network tab in developer tools. It should first load meetings for the past 6 months, and once that's complete start loading meetings for the past 3 years.
  * Try running `make drush-cr` after some page loads and see how it affects the load times
* Check the policymakers list: https://helsinki-paatokset.docker.so/fi/paattajat/
  * The full load times can only be tested on the test environment, but overall it should first load the initial page and then the list, so compared to the test site version 
  * Compare the data to https://nginx-paatokset-test.agw.arodevtest.hel.fi/fi/paattajat
  * If you're missing the "City board sub-committees" accordion, edit this node and toggle the "City council division" under the "Organization label" section: https://helsinki-paatokset.docker.so/fi/paattajat/kaupunginhallituksen-elinkeinojaosto
* Read the code changes to make sure everything makes sense
